design-proposals/helm-chart.md

```diff
@@ -0,0 +1,249 @@
+
+Currently, the manifests in KubeVirt are generated in two steps:
+1. `resource-generator` generates semi-populated configurations under `/manifests/generated`
+2. `manifest-templator` produces the final version of the manifests which are then shipped as part of each GitHub release
+
+### Approach 1: Generate Helm chart(s)
+
+Helm charts are in general not that different from plain Kubernetes manifests.
+It all depends on how much customization we'd allow the user but for the SUSE Edge initial chart we kept as little modifications as possible.
+
+We should be able to plug in the functionality to generate a `Chart.yaml`, `values.yaml` and possibly hooks and organize those in the necessary file structure.
+The charts can then be generated, packaged and pushed to the configured Helm repository at build time.
+
+### Approach 2: Manually curate a Helm chart
+
+We can depend on the Kubernetes manifests provided as part of GitHub releases and create a Helm chart based on those.
+This is the approach I've seen most people attempt when trying to come up with a custom KubeVirt installation method.
+
+It does not involve any changes to the manifest generation flow, however, some degree of automation would still be necessary in order to patch, package and publish the charts.
+The end result probably needs to be kept outside the main KubeVirt repository.
+
+### Overview
+
+In theory, both approaches should work.
+
+[Approach 1](#approach-1-generate-helm-charts) will probably require more effort when it comes to the implementation,
+but it should be stable and both manifests and charts would be released simultaneously.
+
+[Approach 2](#approach-2-manually-curate-a-helm-chart) is the more popular choice,
+however, we need to keep in mind that the complete manifests are not version controlled.
+This would also mean that updated Helm charts are released separately from the manifests.
+
+Unfortunately, I'm still very new to KubeVirt so feedback in this section would be more than welcome.
+
+I'm not familiar with CDI's current manifest creation flow yet, but I *believe* it is similar.
+
+## API Examples
+
+Working with Helm charts is fairly straightforward. Users should be able to manage the applications by:
+
+```shell
+$ helm repo add kubevirt <repo-url>
+$ helm install kubevirt kubevirt/kubevirt --version <version> \
+      # additional configuration
+$ helm install cdi kubevirt/cdi --version <version> \
+      # additional configuration
+$ helm uninstall kubevirt
+$ helm uninstall cdi
+```
+
+## Scalability
+
+N/A
+
+## Update/Rollback Compatibility
+
+The feature will not impact older versions of KubeVirt or CDI.
+Helm charts will, however, be a viable way of updating & rolling back future releases.
+
+## Functional Testing Approach
+
+- Deploy KubeVirt from Helm chart
+- Ensure all components are created and work as expected
+- Slightly modify the configuration of the CRD, CR and operator's Deployment and upgrade KubeVirt
```

Comment:
I think this should be rather "update KubeVirt from N-1 to N release using Helm charts"

Reply:
This is definitely the better wording, will use it. Thanks!